### PR TITLE
Johann social media links

### PR DESCRIPTION
**Pull request by JohannBillar**
The UI for profile social media is done! 🔥  

Next wire it up to redux and have it update the firebase db which in turn updates redux.

### CODE REVIEW

**Comment by JohannBillar** [comment]
🚧  Hold off merging.

**Comment by omarcodex** [comment]
@JohannBillar so awesome! Let's pair on this soon. FYI - I'm noticing that when I add my Twitter data, for example, then re-load the social form, the Twitter data gets moved into the LinkedIn input.

**Comment by JohannBillar** [comment]
@omarcodex Thanks for testing. That bug should be fixed. You might have to hard refresh to update the js bundle in your client. Let me know.

**Comment by JohannBillar** [comment]
@omarcodex and @TinkerDo This branch is ready to be merged! It's up on 🔥 base and ready for testing. Thanks!

**Comment by omarcodex** [comment]
@JohannBillar It's looking good! Nicely done!

I'm thinking next step might be to prevent erroneous inputs, like just "@handle", etc, from going through.

☠️😎

File: src/js/components/Profile.jsx [hunk 1]
```diff
@@ -71,12 +82,20 @@ const Profile = (props: {
...
+            handleSocialUrlChange={handleSocialUrlChange}
           />
         )}
       </div>
     );
   }
-  // otherwise just view a profile
+
+  // Otherwise just view a profile
+  const selectedProfile = filter(appProfiles, profile => profile.uid === uid);
```
Comment (omarcodex) [inline src/js/components/Profile.jsx]: Nice touch!

File: src/js/components/SocialMediaIcons.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  social: {
+    color: '#24292e'
+  }
+};
+
+const SocialMediaIcons = ({ classes, socialUrls }: Object) => {
+  const { facebookUrl, twitterUrl, instagramUrl, linkedInUrl } = socialUrls;
+  return (
+    <div className={classes.social}>
+      {!facebookUrl && <SocialIcon siteName={'fa-facebook-square'} />}
```
Comment (omarcodex) [inline src/js/components/SocialMediaIcons.jsx]: Definitely a good call on the favicons

File: src/js/components/SocialMediaEditModal.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      twitterUrl: this.props.socialInputs.valueTW,
+      instagramUrl: this.props.socialInputs.valueIG,
+      linkedInUrl: this.props.socialInputs.valueIN
+    };
+    this.props.handleSocialMediaUrlSubmit(socialUrls, this.props.uid);
+    this.toggle();
+  };
+
+  validateSocialUrl = (e: Object, site: string) => {
+    const urlRegEx = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/;
```
Comment (omarcodex) [inline src/js/components/SocialMediaEditModal.jsx]: Perhaps here is where we could do extra 'cleaning'

Comment (JohannBillar) [reply src/js/components/SocialMediaEditModal.jsx]: Yes! This is the spot to ad a fancy regexs to only allow profiles, it has an added complication of needing one for each social media type.

File: src/js/components/Profile.jsx (hunk 1 above)
Comment (JohannBillar) [reply src/js/components/Profile.jsx]: Thanks. Hope this doesn't introduce add complexity. Let me know how readable this is for you.

**Pull request merged**